### PR TITLE
Issue #89 feat: Category plugin enhancements

### PR DESCRIPTION
**Pull request by manojLondhe**
- Update get single category api
- Update get category list api

### CODE REVIEW

File: categories/README.md [hunk 1]
```diff
@@ -1,51 +1,57 @@
...
 ## Get Categories List
+
 ```http
 GET /index.php?option=com_api&app=categories&resource=categories&format=raw
 ```
+
+Alternatively, you can call this API as follow (assuming you have created a menu with alias as 'api' for com_api)
+
+```http
+GET /api/categories/categories
```
Comment (coolbung) [inline categories/README.md]: Can this be `/categories/all/:extension`. The value of extension will be available in the `id` param. Not ideal, but the non-sef URL becomes semantic.

File: categories/categories/category.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+		// 2. Then, get an instance of the generic articles model
+		$model = JModelLegacy::getInstance('Category', 'CategoriesModel', array('ignore_request' => true));
+
+		// Set application parameters in model
+		$app   = JFactory::getApplication();
+		$input = $app->input;
+
+		// Important to get from input directly and not from $input->get->get()
+		$id = $input->get('id', 0, 'int');
```
Comment (coolbung) [inline categories/categories/category.php]: Please support getting the category by either id or alias, so both `/categories/category/59` and `/categories/category/sports` should work

Comment (manojLondhe) [reply categories/categories/category.php]: @coolbung 

How do we differentiate between if an alias is set or an id is set?

Consider an example with a 
1. category having `id` as `2` 
2. and another category having `alias` as `sports-2`

Current `com_api` router only handles id, how will it work for alias?

in above two cases:

`$id = $input->get('id', 0, 'int');`  will return `2` as `$id`

Comment (coolbung) [reply categories/categories/category.php]: I can think of 2 options
1. if the id is numeric assume it to be a id, if not alias
2. Support a X-Identifier header. The value for this header can be id (default) or alias. This tells the plugin which record to update. This method is already implemented in the user edit plugin by @twsvaishali

Comment (manojLondhe) [reply categories/categories/category.php]: 3. or can we add trigger to let extend com_api router, something like

```onApiRoute (&segments, &$vars)```

and let plugins set `vars` as needed

I tried this out and I got both id and alias to work

File: categories/README.md (hunk 1 above)
Comment (manojLondhe) [reply categories/README.md]: need router support here as well as described at https://github.com/techjoomla/com_api-plugins/pull/90/files#r245647296 @coolbung

**Pull request merged**